docs/proposals/backend-support-appset.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Adding backend support for ApplicationSets RBAC CLI/Web requests
+authors:
+  - "@jgwest" # Authors' github accounts here.
+sponsors:
+  - TBD        # List all intereste parties here.
+reviewers:
+  - "@alexmt"
+  - "@jannis"
+approvers:
+  - "@alexmt"
+  - "@jannis"
+
+creation-date: 2021-04-05
+last-updated: 2021-04-05
+---
+
+# Neat Enhancement Idea
+
+This is a proposal to add support for creating ApplicationSets via the Argo CD Web/CLI, by adding support to the ApplicationSet and API Server backend that respects Argo CD RBAC.
+
+## Summary
+
+Currently, users can only create ApplicationSets by applying configurations declaratively using `kubectl`. Introducing support for CLI/UI would improve the overall experience for managing large number of applications using ApplicationSet.
+
+## Motivation
+
+As ApplicationSet Controller is now part of ArgoCD installation, we would like to allow users to be able to create/update/delete application sets via CLI/UI.
+
+### Goals
+
+* **Expose endpoints in the API server to interact with ApplicationSets**
+
+  Users should be able to create/update/delete ApplicationSets using argocd CLI/UI.
+
+* **Changes to CLI**
+
+  Add a new argocd command option `appset` with `create`, `delete` and `update` sub-commands.
+
+## Proposal
+
+This is the high level overview of creation (update/deletion) of an ApplicationSet via Argo CD CLI/UI.
+
+![High Level Architecture](./backend-support-appset.png)
+
+1. User issues `argocd appset create/update/delete`(or Web UI equivalent) command from CLI, into an Argo CD server on which they are logged-in. The command converts the command request into GRPC and sends it off to Argo CD API Server.
+
+#### **Argo CD API Server:**
+
+2. The API Server receives Create/Update/Delete request via GRPC and verifies that ApplicationSet controller is installed within the namespace (if not, return an error response back to user). 
```

Comment:
IMO, if we do not throw a proper error in response to cli call, it might not be a good user experience.

Reply:
This step is less necessary now that ApplicationSet controller is a mandatory install of Argo CD. When the proposal was written, it was a separate project.

So it's fine for ApplicationSet to just attempt to send the GRPC request to the appset controller, and report an error back to the user if it can't.